osu.Game/Screens/Edit/Editor.cs

```diff
@@ -703,11 +712,53 @@ private List<MenuItem> createFileMenuItems()
             if (RuntimeInfo.IsDesktop)
                 fileMenuItems.Add(new EditorMenuItem("Export package", MenuItemType.Standard, exportBeatmap));
 
+            fileMenuItems.Add(new EditorMenuItemSpacer());
+
+            var beatmapSet = beatmapManager.QueryBeatmapSet(bs => bs.ID == Beatmap.Value.BeatmapSetInfo.ID) ?? playableBeatmap.BeatmapInfo.BeatmapSet;
+
+            var difficultyItems = new List<MenuItem>();
+
+            foreach (var rulesetBeatmaps in beatmapSet.Beatmaps.GroupBy(b => b.RulesetID).OrderBy(group => group.Key))
+            {
+                if (difficultyItems.Count > 0)
+                    difficultyItems.Add(new EditorMenuItemSpacer());
+
+                foreach (var beatmap in rulesetBeatmaps.OrderBy(b => b.StarDifficulty))
+                    difficultyItems.Add(createDifficultyMenuItem(beatmap));
+            }
+
+            fileMenuItems.Add(new EditorMenuItem("Change difficulty") { Items = difficultyItems });
+
             fileMenuItems.Add(new EditorMenuItemSpacer());
             fileMenuItems.Add(new EditorMenuItem("Exit", MenuItemType.Standard, this.Exit));
             return fileMenuItems;
         }
 
+        private DifficultyMenuItem createDifficultyMenuItem(BeatmapInfo beatmapInfo)
+        {
+            bool isCurrentDifficulty = playableBeatmap.BeatmapInfo.Equals(beatmapInfo);
+            return new DifficultyMenuItem(beatmapInfo, isCurrentDifficulty, switchToDifficulty);
+        }
+
+        private void switchToDifficulty(BeatmapInfo beatmapInfo)
+        {
+            if (loader == null)
+                return;
+
+            loader.ValidForResume = true;
+            this.Exit();
+            loader.ScheduleDifficultySwitch(beatmapInfo);
```

Comment:
This process looks weird to me. Anything against a single method in the loader which calls `this.MakeCurrent()` and sets its state correctly?

Reply:
Sounds reasonable. I'll try it and see if any problems fall out.

Reply:
This flow should be closer to expectations with the latest changes, I think.